server/src/main/java/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -2083,8 +2084,15 @@
         if (clusterId != null) {
             sc.setParameters("clusterId", clusterId);
         }
-        // Don't return DomR and ConsoleProxy volumes
-        sc.setParameters("type", VirtualMachine.Type.ConsoleProxy, VirtualMachine.Type.SecondaryStorageVm, VirtualMachine.Type.DomainRouter);
+
+        sc.setParameters("type", VirtualMachine.Type.User);
+
+        // Display all volumes for ROOT admin
+        if (forSystemVms != null && forSystemVms && caller.getType() == Account.ACCOUNT_TYPE_ADMIN) {
```

Comment:
What about extracting this condition into a new method, such as:
````
boolean isAdminRequestingAllVolumes(Boolean forSystemVms, Account caller) {
   return forSystemVms != null && forSystemVms && caller.getType() == Account.ACCOUNT_TYPE_ADMIN;
}
````

Reply:
I still need `sc.setParameters("systemUse", 0, 1);` and others can be avoided

Reply:
+1 on @nvazquez suggestions, it looks like a simple way of improving the code.